tsl/src/nodes/decompress_chunk/pred_vector_array.c

```diff
@@ -47,7 +47,7 @@ vector_array_predicate_impl(VectorPredicate *vector_const_predicate, bool is_or,
 	char typalign;
 	get_typlenbyvalalign(ARR_ELEMTYPE(arr), &typlen, &typbyval, &typalign);
 
-	const char *array_data = (const char *) ARR_DATA_PTR(arr);
+	const char *restrict array_data = (const char *) ARR_DATA_PTR(arr);
```

Comment:
Could you add a comment on which compiler optimizations we are getting from the `restrict` type qualifier?

Reply:
I just add `restrict` to every piece of data that is used in hot loops and lives on the heap. The general idea is to tell the compiler that they point to different chunks of memory, so that more efficient code can be generated. It's important for the data you write, otherwise technically everything has to be re-read for a new loop iteration, because it might have been modified by a write. I think it's not really important for read-only variables, but I find it simpler to stick it on everything to avoid the need of deeper analysis which can be error-prone. So this is the general idea, I think spelling this out in the code comments each time would be too verbose and not very useful.

Reply:
>  I think it's not really important for read-only variables, but I find it simpler to stick it on everything to avoid the need of deeper analysis which can be error-prone.

Maybe I should use `const` for read-only objects and `restrict` for read/write.

Reply:
https://github.com/timescale/timescaledb/pull/6791